localstack/services/s3/notifications.py

```diff
@@ -136,7 +136,7 @@ def from_request_context(
             )
         except NoSuchKey as ex:
             if allow_non_existing_key:
-                key: FakeKey = FakeKey(key_name, "")
+                key: FakeKey = FakeKey(key_name, "", request_context.account_id)
```

Comment:
they finally fixed the key not having an account id in multipart parts, so that the S3Backend would be initialized with no account id, which led to an issue with persistence. Better late than never I guess!

Reply:
> Better late than never I guess!

This was changed in Moto to improve the type annotations, @bentsku . It wasn't fixed as a bug, because no-one has ever raised an issue/PR to indicate there was a problem with it.

Reply:
Ah, sorry: my tone was not great re-reading myself.
This bug was mostly on our side with the persistence mechanism, the multipart `Part` object not having an account id was not really causing a bug in Moto itself, so we fixed our handling of it. It is true that I could have opened a bug report about it, but it felt very specific and not adding anything to Moto. Reading my comment, I can see how rude it sounds. My apologies.

Reply:
All good, no harm done. Appreciate the explanation!